content/sensu-core/0.29/guides/scaling-overview.md

```diff
@@ -0,0 +1,115 @@
+
+# Sensu Components
+
+A typical Sensu deployment consists of four pieces:
+
+* `sensu-server`
+* `sensu-api`
+* RabbitMQ (message bus)
+* Redis (data store)
+
+There can be variation when it comes to the message bus and data store components, but using Redis as the data store and RabbitMQ as the message bus is the most common (and supported) way of deploying those deployment components.
+
+## Sensu Server
+
+The `sensu-server` process is the workhorse of any deployment. It performs a [number of tasks][1] including check scheduling and publishing, monitoring clients via keepalives, and event processing. To scale this component, you need only add the desired number of Sensu servers and point them at your RabbitMQ instance where they'll do their own internal master election.
+
+## Sensu API
+
+The `sensu-api` component is a stateless http frontend. It can be scaled with traditional http load-balancing strategies (HAproxy, Nginx, etc.). Each additional API instance will need to be configured to point to your Redis instance, and can then be added to your load balancing pool.
+
+## Redis
+
+Redis can be scaled out in several different ways. Using Redis Sentinel is the primary supported way of scaling Redis. You can read more about installing and configuring Sentinel in our [Redis reference documentation][2].
+
+## RabbitMQ
+
+RabbitMQ can be used in a clustered configuration for Sensu. You can read more about configuring RabbitMQ clusters in our [RabbitMQ reference documentation][3].
+
+## Scaling Sensu at a Single Site
+
+Each Sensu component can be scaled independently at a single site, whether you need to scale out the number of consumers (`sensu-server` instances) to keep your RabbitMQ queue depth to manageable level, or you need to ensure that Redis is highly available. We'll put all of these elements together in the next guide.
+
+Keep reading for more information about multi-site scaling strategies.
+
+## Scaling Sensu Across Multiple Sites {#scaling-sensu-across-multiple-sites}
+
+Every distributed system, Sensu included, must take into account special considerations when scaling across multiple sites (datacenters) where the networking (WAN) will be unreliable.
+
+For the purpose of this documentation each site will be referred to as a "datacenter".
+
+### Strategy 1: Isolated Clusters Aggregated by Uchiwa
+
+This strategy involves building isolated, independent Sensu server/clusters at each datacenter, and then using Uchiwa's multi-datacenter configuration option to get an aggregate view of the events and clients.
+
+#### Pros
+
+* WAN instability does *not* lead to flapping Sensu checks
+* Sensu operation continues un-interrupted during a WAN outage
+* The overall architecture is easier to understand and troubleshoot
+
+#### Cons
+
+* WAN outages mean a whole Datacenter can go dark and not set off alerts (cross-datacenter checks are therefor essential)

```

Comment:
Maybe I am not following this correctly, but you list a Con here where a `Datacenter can go dark and not set off alerts`, but a Pro above is `Sensu operation continues un-interrupted during a WAN outage `. These seem to contradict each other a bit.

Feels like the con should be that a whole Datacenter can go dark in Uchiwa, giving no dashboard visibility. That said, depending on the outage, it could mean that the Datacenter can't alert on itself either, but it is possible that just your centralized Uchiwa dashboard can't reach the API of the other data center(s), but those other datacenter(s) can still reach the internet, which would allow events to still be handled and be sent out.

Reply:
Good point. That's pulled from the original doc here https://github.com/sensu/sensu-docs-legacy/blob/master/legacy/0.16/scaling_strategies.md , so maybe we're both lacking a bit of context? Not sure if @cwjohnston would have any additional context for it, or could interpret.